plugins/catalog-backend/src/service/AuthorizedNextEntitiesCatalog.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { catalogEntityReadPermission } from '@backstage/plugin-catalog-common';
+import {
+  AuthorizeResult,
+  PermissionAuthorizer,
+} from '@backstage/plugin-permission-common';
+import { ConditionTransformer } from '@backstage/plugin-permission-node';
+import {
+  EntitiesCatalog,
+  EntitiesRequest,
+  EntitiesResponse,
+  EntityAncestryResponse,
+  EntityFilter,
+} from '../catalog/types';
+
+export class AuthorizedNextEntitiesCatalog implements EntitiesCatalog {
```

Comment:
I think `AuthorizedEntitiesCatalog` is a more fitting name perhaps? There's nothing `Next` about this x)

Reply:
Done!